server/src/core/server/services/tenant/disposableEmails.ts

```diff
@@ -0,0 +1,53 @@
+import axios from "axios";
+import Logger from "bunyan";
+import { Parser } from "stream-json";
+import { streamArray } from "stream-json/streamers/StreamArray";
+
+import {
+  DISPOSABLE_EMAIL_DOMAIN_CACHE_DURATION,
+  DISPOSABLE_EMAIL_DOMAINS_LIST_URL,
+  DISPOSABLE_EMAIL_DOMAINS_REDIS_KEY,
+} from "coral-common/common/lib/constants";
+
+import { AugmentedRedis } from "../redis";
+
+export async function readDisposableEmailDomainsAndAddToRedis(
+  redis: AugmentedRedis,
+  logger: Logger
+) {
+  const now = new Date();
+
+  try {
+    const stream = await axios.get(`${DISPOSABLE_EMAIL_DOMAINS_LIST_URL}`, {
+      responseType: "stream",
+    });
+
+    const pipeline = stream.data.pipe(new Parser()).pipe(streamArray());
+
+    pipeline.on("data", async (data: { value: string }) => {
+      const domain = data.value;
+      await redis.set(
```

Comment:
I think you'll want to use a hash set lookup for these values instead of a pipeline:
https://redis.io/docs/latest/commands/hset/

Also, if you're going to call `set` many, many times, it's better to use mset instead of pipelining multiple commands, it's orders of magnitude faster.

Reply:
thanks for the note/links on this! I updated to use the `hset` here, and it now gets wiped before being updated each time.